extra_data/components.py

```diff
@@ -983,29 +1105,36 @@ def _make_image_index(self, tids, inner_ids, inner_name='pulse'):
         )
 
 
+@detectors
 class JUNGFRAU(MultimodDetectorBase):
     """An interface to JUNGFRAU data.
 
+    JNGFR, JF1M, JF4M all store data in a "data" group, with trains along
+    the first and memory cells along the second dimension.
+    This allows only a set number of frames to be stored for each train.
+
     Parameters
     ----------
 
     data: DataCollection
       A data collection, e.g. from :func:`.RunDirectory`.
-    modules: set of ints, optional
-      Detector module numbers to use. By default, all available modules
-      are used.
     detector_name: str, optional
       Name of a detector, e.g. 'SPB_IRDA_JNGFR'. This is only needed
       if the dataset includes more than one JUNGFRAU detector.
+    modules: set of ints, optional
+      Detector module numbers to use. By default, all available modules
+      are used.
     min_modules: int
       Include trains where at least n modules have data. Default is 1.
     """
     # We appear to have a few different formats for source names:
     # SPB_IRDA_JNGFR/DET/MODULE_1:daqOutput  (e.g. in p 2566, r 61)
     # SPB_IRDA_JF4M/DET/JNGFR03:daqOutput    (e.g. in p 2732, r 12)
     # FXE_XAD_JF1M/DET/RECEIVER-1
+    n_modules = 8
```

Comment:
I'd like to avoid hardcoding this, if possible. JUNGFRAU detectors are built from varying numbers of modules - we happen to be looking first at a 4MPx configuration with 8 modules, but the `JF1M` name in the comments suggests we have a 2 module setup, and the [JUNGFRAU description](https://www.psi.ch/en/detectors/jungfrau) mentions 1, 3, 9 and 32 module installations at other facilities.

I think we should either detect the number of modules from the matching sources when the class is created, or let the user specify how many modules they expect in the data. For JUNGFRAU geometry in EXtra-geom, we rely on user input to know how many modules are present:

https://github.com/European-XFEL/EXtra-geom/blob/ac76cc5d71f1f7bb7510f08fdf9f3092dbda3829/extra_geom/detectors.py#L2025-L2028

Reply:
Implemented in https://github.com/European-XFEL/EXtra-data/commit/156b8bf353b59b107950a2baf566352c061acc4c (not used in 'write_cxi.py' just yet).